server/test.js

```diff
@@ -0,0 +1,29 @@
+"use strict";
+
+var request = require("request");
+
+var host = "localhost";
+if(process.env.DOCKER_HOST) {
+  // Get the IP out of the DOCKER HOST if this machine is using docker machine or boot2docker.
+  var rx = /(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)/;
+  host = rx.exec(process.env.DOCKER_HOST)[0];
+}
+var port = process.env.PORT || 8000;
+
+var uri = "http://" + host + ":" + port + "/tile";
```

Comment:
Ick.  This is why I was suggesting that the test start up its own instance of the service on a domain socket (or a port that you're certain won't already be in use).

To facilitate this, `server/server.js` would be split into the Express app and the boilerplate that wires up the app and starts the server (that's the bit that would be replaced in this test).